packages/oidc-client/src/lib/client.store.ts

```diff
@@ -207,29 +261,77 @@ export async function oidc<ActionType extends ActionTypes = ActionTypes>({
           const err = {
             error: 'No access token found',
             type: 'auth_error',
-          } as AuthorizeErrorResponse;
+          } as const;
 
           log.error(err.error);
 
           return err;
         }
 
-        return await store.dispatch(
-          oidcApi.endpoints.userInfo.initiate({
-            accessToken: tokens.accessToken,
-            endpoint: wellknown.userinfo_endpoint,
+        const info = Micro.promise(() =>
+          store.dispatch(
+            oidcApi.endpoints.userInfo.initiate({
+              accessToken: tokens.accessToken,
+              endpoint: wellknown.userinfo_endpoint,
+            }),
+          ),
+        ).pipe(
+          Micro.map(({ data, error }) => {
+            if (error) {
+              let message = 'An error occurred while fetching user info';
+              let status: number | string = 'unknown';
+              if ('message' in error && error.message) {
+                message = error.message;
+              }
+              if ('status' in error) {
+                status = error.status;
+              }
+              return {
```

Comment:
I am curious if we added `readonly _tag` property if we would get the benefit of Error discrimination without using the `Micro.TaggedError` class